Spec/RealtimeClientPresence.swift

```diff
@@ -77,6 +77,59 @@ class RealtimeClientPresence: QuickSpec {
                 }
             }
 
+            // RTP2
+            pending("should be used a PresenceMap to maintain a list of members") {
+                let options = AblyTests.commonAppSetup()
+                var clientSecondary: ARTRealtime!
+                defer { clientSecondary.close() }
+
+                waitUntil(timeout: testTimeout) { done in
+                    clientSecondary = AblyTests.addMembersSequentiallyToChannel("test", members: 100, options: options) {
+                        done()
+                    }.first
+                }
+
+                let client = AblyTests.newRealtime(options)
+                defer { client.close() }
+                let channel = client.channels.get("test")
+
+                channel.presence.subscribe { presence in
+                    expect(presence.action).to(equal(ARTPresenceAction.Leave))
+                    expect(presence.clientId).to(equal("user50"))
+                }
+
+                channel.presenceMap.testSuite_getArgumentFrom("put:", atIndex: 0) { arg0 in
+                    let member = arg0 as! ARTPresenceMessage
+                    if member.clientId == "user50" {
+                        print(member.timestamp)
+                    }
+                }
+
+                waitUntil(timeout: testTimeout) { done in
+                    channel.attach() { _ in
+                        let transport = client.transport as! TestProxyTransport
+                        transport.beforeProcessingReceivedMessage = { protocolMessage in
+                            // A leave event for a member can arrive before that member is later registered as present as part of the initial SYNC operation.
```

Comment:
Agreed, I think the only viable way to test this is to send a mock protocol message to the library mid sync before `user50` has been received simulating that message being received.